native_libs/src/Core/ArrowUtilities.h

```diff
@@ -738,6 +742,50 @@ DFH_EXPORT void validateIndex(const arrow::Array &array, int64_t index);
 DFH_EXPORT void validateIndex(const arrow::ChunkedArray &array, int64_t index);
 DFH_EXPORT void validateIndex(const arrow::Column &column, int64_t index);
 
+
+template<arrow::Type::type id1, arrow::Type::type id2, typename F>
+void iterateOverPairs(const arrow::ChunkedArray &array1, const arrow::ChunkedArray &array2, F &&f)
+{
+    assert(array1.length() == array2.length());
+    const auto N = array1.length();
+
+    auto chunks1Itr = array1.chunks().begin();
+    auto chunks2Itr = array2.chunks().begin();
+
+    int64_t row = 0;
+
+    int32_t chunk1Length = (int32_t)(*chunks1Itr)->length(); // arrow specification says that array size is 32-bit
```

Comment:
If size is 32-bit shouldn't this be `uint_32t`?

Reply:
As per Apache Arrow specification, array's length is representible by 32-bit signed integer. See https://arrow.apache.org/docs/format/Layout.html#array-lengths 
So we are fine. As for why signed instead of unsigned when both works:
* Apache Arrow uses signed integers for length everywhere
* there is a growing sentiment in C++ community (which I share) that generally using unsigned integers for representing collection lengths wasn't that good idea after all. ( https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-nonnegative )

Reply:
Fair! Thanks for the info!